logstash-core/src/main/java/org/logstash/instrument/witness/process/ProcessWitness.java

```diff
@@ -0,0 +1,247 @@
+            openFileDescriptors.set(currentOpen);
+            if (maxFileDescriptors.getValue() == null || peakOpenFileDescriptors.getValue().longValue() < currentOpen) {
+                peakOpenFileDescriptors.set(currentOpen);
+            }
+            maxFileDescriptors.set(unixOsBean.getMaxFileDescriptorCount());
+        }
+        cpu.refresh();
+        memory.refresh();
+    }
+
+    /**
+     * Get a reference to associated snitch to get discrete metric values.
+     *
+     * @return the associate {@link Snitch}
+     */
+    public Snitch snitch() {
+        return snitch;
+    }
+
+    /**
+     * An inner witness for the process / cpu metrics
+     */
+    public class Cpu implements ScheduledWitness {
+        private final static String KEY = "cpu";
+        private final NumberGauge cpuProcessPercent;
+        private final NumberGauge cpuTotalInMillis;
+
+        private Cpu() {
+            this.cpuProcessPercent = new NumberGauge("percent", -1);
+            this.cpuTotalInMillis = new NumberGauge("total_in_millis", -1);
+        }
+
+        @Override
+        public void refresh() {
+            cpuProcessPercent.set(scaleLoadToPercent(unixOsBean.getProcessCpuLoad()));
+            cpuTotalInMillis.set(TimeUnit.MILLISECONDS.convert(unixOsBean.getProcessCpuTime(), TimeUnit.NANOSECONDS));
+        }
+    }
+
+    /**
+     * An inner witness for the the process / memory metrics
+     */
+    public class Memory implements ScheduledWitness {
+        private final static String KEY = "mem";
+        private final NumberGauge memTotalVirtualInBytes;
+
+        private Memory() {
+            memTotalVirtualInBytes = new NumberGauge("total_virtual_in_bytes", -1);
+        }
+
+        @Override
+        public void refresh() {
+            memTotalVirtualInBytes.set(unixOsBean.getCommittedVirtualMemorySize());
+        }
+    }
+
+    @Override
+    public void genJson(JsonGenerator gen, SerializerProvider provider) throws IOException {
+        SERIALIZER.innerSerialize(this, gen, provider);
+    }
+
+    /**
+     * The Jackson serializer.
+     */
+    static class Serializer extends StdSerializer<ProcessWitness> {
+        /**
+         * Default constructor - required for Jackson
+         */
+        public Serializer() {
+            this(ProcessWitness.class);
+        }
+
+        /**
+         * Constructor
+         *
+         * @param t the type to serialize
+         */
+        protected Serializer(Class<ProcessWitness> t) {
+            super(t);
+        }
+
+        @Override
+        public void serialize(ProcessWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeStartObject();
+            innerSerialize(witness, gen, provider);
+            gen.writeEndObject();
+        }
+
+        void innerSerialize(ProcessWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
```

Comment:
@jakelandis this method is `static` isn't it?

Reply:
Yeah there are few things minor things in this review don't mesh with some of the clean up you did and I reviewed this AM.  I will take a quick sweep through and clean it up.